src/Core/Package.vala

```diff
@@ -367,19 +367,28 @@ public class AppCenterCore.Package : Object {
         return name;
     }
 
-    public string? get_description () {
+    public async string? get_description () {
+        SourceFunc callback = get_description.callback;
+
         if (description != null) {
-            return description;
+            Idle.add((owned)callback);
         }
 
-        description = component.get_description ();
-        if (description == null) {
-            var package = find_package ();
-            if (package != null) {
-                description = package.description;
+        new Thread<bool> ("appstream_description", () => {
+            description = component.get_description ();
```

Comment:
I think you can ommit getting the description from `AppStream.Component` in a thread, since it's just a property get while `find_package` is the call that actually blocks the UI.

Reply:
I'll try that out.

Reply:
Finished. Commit coming soon.

Reply:
What I meant is more something like this: 
```
description = component.get_description ();
if (description != null) {
    return description;
}

new Thread<bool> (... () => { ... });
```

Reply:
Will submit a change for that soon.

Reply:
Done.